service/controller/appcatalog/v1/resource/index/desired.go

```diff
@@ -0,0 +1,57 @@
+package index
+
+import (
+	"context"
+	"io/ioutil"
+	"k8s.io/api/core/v1"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/giantswarm/microerror"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/pkg/label"
+	"github.com/giantswarm/app-operator/service/controller/appcatalog/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	storageURL, err := url.Parse(key.AppCatalogStorageURL(cr))
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	storageURL.Path = path.Join(storageURL.Path, "index.yaml")
+	response, err := http.Get(storageURL.String())
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if response.StatusCode != 200 {
+		return nil, microerror.Mask(indexNotFound)
+	}
+
+	defer response.Body.Close()
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	configMap := &v1.ConfigMap{
+		ObjectMeta: v12.ObjectMeta{
+			Name:        cr.Name,
+			Namespace:   cr.Namespace,
+			Labels:      label.ProcessLabels(r.projectName, "", cr.ObjectMeta.Labels),
```

Comment:
do we really replicate all labels of the CR to the CM? Could there be some that are not right here, like managed-by or sth?

Reply:
The purpose of `ProcessLabels` is to get this right. There are some like the `managed-by` label that we need to update.

There are others like `giantswarm.io/cluster` and `giantswarm.io/organization` that are useful to pass on. Then we can just set them once in cluster-operator or wherever. 

But yes we should try and keep this simple.

Reply:
I'm also currently thinking if the whole label stuff would not be a case for OPA policies, but ok for now